### PR TITLE
Add simulated device readme.

### PR DESCRIPTION
**Pull request by krypton36**
#### Problem
There currently is no readme on how to generate, build, run, interact, and add simulated device tests.

#### Change overview
Add a new readme that informs users on how to generate, build, run, interact and add simulated device test.

#### Testing
N/A

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking documentation changes. Still requires restyle though.

**Review by trombadore-google — APPROVED** [review]

File: docs/guides/simulated_device_linux.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+added. It is defined by a [ZAP config]() file and tests can be added with a
+[YAML file](https://github.com/project-chip/connectedhomeip/tree/master/src/app/tests/suites/certification/Test_TC_DM_1_3_Simulated.yaml).
+<br><br>
+
+### Prerequisite
+
+<hr>
+
+-   [Building Prerequisites](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/BUILDING.md#prerequisitesn)
+-   [Prepare For Bduiling](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/BUILDING.md#prepare-for-building)
```
Comment (woody-apple) [inline docs/guides/simulated_device_linux.md:28]:
```suggestion
-   [Prepare For Building](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/BUILDING.md#prepare-for-building)
```

File: docs/guides/simulated_device_linux.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+## Running the app with test parameter
+
+Now that the building is completed there is a `chip-app1` binary created. This
+binary can be executed on a linux os with test commands.
+
+1. To generate the ZAP files, and build the `chip-app1` binary completing the
+   following steps:
+
+    ```
+    ./out/simulated/chip-app1 --command [TEST MAME]
```
Comment (woody-apple) [inline docs/guides/simulated_device_linux.md:86]:
```suggestion
    ./out/simulated/chip-app1 --command [TEST NAME]
```

File: docs/guides/simulated_device_linux.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+1. YAML test file are located in
+   [YAML folder](https://github.com/project-chip/connectedhomeip/tree/master/src/app/tests/suites/certification/)
+2. Test names must follow a strict format dues to CI of test recognition. The
+   format is as follows:
+    - Test_TC\_[`CATEGORY ABBREVIATION`]\_[`SECTION NUMBER`]\_[`SUBSECTION
+      NUMBER`]\_Simulated.yaml
+    - <strong>`IMPORTANT`: The test name must end in Simulated with the
+      capital.</strong>
+3. Available properties can be found in
+   [YAML Test Reame](https://github.com/project-chip/connectedhomeip/tree/master/src/app/tests/suites/README.md)
```
Comment (woody-apple) [inline docs/guides/simulated_device_linux.md:133]:
```suggestion
   [YAML Test Name](https://github.com/project-chip/connectedhomeip/tree/master/src/app/tests/suites/README.md)
```

File: docs/guides/simulated_device_linux.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    ./out/debug/standalone/chip-tool onoff on 0x654321 1
+    ./out/debug/standalone/chip-tool onoff read on-off 0x654321 1
+    ./out/debug/standalone/chip-tool onoff write on-time 1 0x654321 1
+    ```
+
+    See
+    [chip-tool readme](https://github.com/project-chip/connectedhomeip/tree/master/examples/chip-tool)
+    for additional commands.
+
+## Adding simulated Tests via TAML
```
Comment (woody-apple) [inline docs/guides/simulated_device_linux.md:118]:
```suggestion
## Adding simulated Tests via YAML
```

File: docs/guides/simulated_device_linux.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+5. [Test_TC_DM_1_3_Simulated](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/certification/Test_TC_DM_1_3_Simulated.yaml)
+   is an example of a written test that runs on the simulated device.
+6. Next, it will need to be added to
+   [examples/placeholder/linux/apps/app1/tests.js](https://github.com/project-chip/connectedhomeip/tree/master/examples/placeholder/linux/apps/app1/tests.js).
+   in the following array
+    ```javascript
+    const tests = ["Test_TC_DM_1_3_Simulated"];
+    ```
+7. Then, the code will beed to generated using ZAP. Follow
```
Comment (woody-apple) [inline docs/guides/simulated_device_linux.md:148]:
```suggestion
7. Then, the code will be generated using ZAP. Follow
```

File: docs/guides/simulated_device_linux.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    ```
+     ./scripts/examples/gn_build_test_example.sh app1
+    ```
+
+## Build the App only
+
+In order to utilize the app against a commissioner or controller, the app will
+need to be specifically built.
+
+1. To only buil the `chip-app1` binary completing the following steps:
```
Comment (woody-apple) [inline docs/guides/simulated_device_linux.md:51]:
```suggestion
1. To only build the `chip-app1` binary completing the following steps:
```

File: docs/guides/simulated_device_linux.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+<br><br>Simulated Device: simulation of an application in which tests can be
+added. It is defined by a [ZAP config]() file and tests can be added with a
+[YAML file](https://github.com/project-chip/connectedhomeip/tree/master/src/app/tests/suites/certification/Test_TC_DM_1_3_Simulated.yaml).
+<br><br>
+
+### Prerequisite
+
+<hr>
+
+-   [Building Prerequisites](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/BUILDING.md#prerequisitesn)
```
Comment (woody-apple) [inline docs/guides/simulated_device_linux.md:27]:
```suggestion
-   [Building Prerequisites](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/BUILDING.md#prerequisites)
```

**Review by woody-apple — APPROVED** [review]

File: docs/guides/simulated_device_linux.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+### Prerequisite
+
+<hr>
+
+-   [Building Prerequisites](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/BUILDING.md#prerequisites)
+-   [Prepare For Building](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/BUILDING.md#prepare-for-building)
+
+<hr>
+
+<a name="virtualization"></a>
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:32]: Are those HTML tags needed?
```suggestion

Simulated Device: simulation of an application in which tests can be
added. It is defined by a [ZAP config]() file and tests can be added with a
[YAML file](https://github.com/project-chip/connectedhomeip/tree/master/src/app/tests/suites/certification/Test_TC_DM_1_3_Simulated.yaml).

### Prerequisite

-   [Building Prerequisites](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/BUILDING.md#prerequisites)
-   [Prepare For Building](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/BUILDING.md#prepare-for-building)

```

File: docs/guides/simulated_device_linux.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+## Generating, and Building the default Simulated App with Script
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:34]:
```suggestion
## Generating and building the default Simulated App with Script
```

File: docs/guides/simulated_device_linux.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
+# Simulated Device How-To (Linux)
+
+This README contains instructions on how to build, run, and interact with a
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:3]:
```suggestion
This document contains instructions on how to build, run, and interact with a
```

File: docs/guides/simulated_device_linux.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+If some parameters need to be overridden, a `CHIPProjectConfig.h` file can be
+placed under an ‘include’ folder into the app folder. For example
+`examples/placeholder/linux/apps/app1/include/CHIPProjectConfig.h`
+
+In order to generate specific tests for a given accessory, a
+[examples/placeholder/linux/apps/app1/tests.js](https://github.com/project-chip/connectedhomeip/tree/master/examples/placeholder/linux/apps/app1/tests.js)
+file can be added into the application directory. The tests listed there are the
+one that will be executed once the application has been commissioned.
+<br><br>Simulated Device: simulation of an application in which tests can be
+added. It is defined by a [ZAP config]() file and tests can be added with a
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:19]: The reference for ZAP config file wasn't provided 
```suggestion
added. It is defined by a ZAP config file and tests can be added with a
```

File: docs/guides/simulated_device_linux.md [hunk 12]
```diff
@@ -0,0 +1,152 @@
+# Simulated Device How-To (Linux)
+
+This README contains instructions on how to build, run, and interact with a
+simulated device. All virtual accessories live in
+[examples/placeholder/linux/apps](https://github.com/project-chip/connectedhomeip/tree/master/examples/placeholder/linux/apps).
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:5]: IMHO, It's better to use relative references
```suggestion
[examples/placeholder/linux/apps](../../examples/placeholder/linux/apps).
```

File: docs/guides/simulated_device_linux.md [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+
+Each accessory needs to be hosted into a subfolder. It will be the name of the
+application. For example `app1` will create a binary named `chip-app1`.
+
+If some parameters need to be overridden, a `CHIPProjectConfig.h` file can be
+placed under an ‘include’ folder into the app folder. For example
+`examples/placeholder/linux/apps/app1/include/CHIPProjectConfig.h`
+
+In order to generate specific tests for a given accessory, a
+[examples/placeholder/linux/apps/app1/tests.js](https://github.com/project-chip/connectedhomeip/tree/master/examples/placeholder/linux/apps/app1/tests.js)
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:15]: ditto line 5
```suggestion
[examples/placeholder/linux/apps/app1/tests.js](../../examples/placeholder/linux/apps/app1/tests.js)
```

File: docs/guides/simulated_device_linux.md [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+file can be added into the application directory. The tests listed there are the
+one that will be executed once the application has been commissioned.
+<br><br>Simulated Device: simulation of an application in which tests can be
+added. It is defined by a [ZAP config]() file and tests can be added with a
+[YAML file](https://github.com/project-chip/connectedhomeip/tree/master/src/app/tests/suites/certification/Test_TC_DM_1_3_Simulated.yaml).
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:20]: ditto line 5
```suggestion
[YAML file](../../src/app/tests/suites/certification/Test_TC_DM_1_3_Simulated.yaml).
```

File: docs/guides/simulated_device_linux.md [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+<br><br>
+
+### Prerequisite
+
+<hr>
+
+-   [Building Prerequisites](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/BUILDING.md#prerequisites)
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:27]: ditto line 5
```suggestion
-   [Building Prerequisites](./BUILDING.md#prerequisites)
```

File: docs/guides/simulated_device_linux.md [hunk 16]
```diff
@@ -0,0 +1,152 @@
...
+-   [Prepare For Building](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/BUILDING.md#prepare-for-building)
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:28]: ditto line 5
```suggestion
-   [Prepare For Building](./BUILDING.md#prepare-for-building)
```

File: docs/guides/simulated_device_linux.md [hunk 17]
```diff
@@ -0,0 +1,152 @@
...
+    ./out/simulated/chip-app1 --command [TEST NAME]
+    ```
+
+## Interacting with the simulated app
+
+Now that the building the app and starting it is complete, you will be able to
+interact with it using chip-tool
+
+1. Follow the instruction to build chip-tool in the
+   [chip-tool readme](https://github.com/project-chip/connectedhomeip/tree/master/examples/chip-tool).
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:95]: ditto line 5
```suggestion
   [chip-tool readme](../../examples/chip-tool).
```

File: docs/guides/simulated_device_linux.md [hunk 18]
```diff
@@ -0,0 +1,152 @@
...
+   chip-tool as follow.
+
+    ```
+    ./out/debug/standalone/chip-tool onoff on 0x654321 1
+    ./out/debug/standalone/chip-tool onoff read on-off 0x654321 1
+    ./out/debug/standalone/chip-tool onoff write on-time 1 0x654321 1
+    ```
+
+    See
+    [chip-tool readme](https://github.com/project-chip/connectedhomeip/tree/master/examples/chip-tool)
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:115]: ditto line 5
```suggestion
    [chip-tool readme](../../examples/chip-tool)
```

File: docs/guides/simulated_device_linux.md [hunk 19]
```diff
@@ -0,0 +1,152 @@
...
+    for additional commands.
+
+## Adding simulated Tests via YAML
+
+In order to validate commissioner/controller behavior, tests need to be added to
+the simulated device test framework. To achieve this, YAML files are created and
+new code is generated.
+
+1. YAML test file are located in
+   [YAML folder](https://github.com/project-chip/connectedhomeip/tree/master/src/app/tests/suites/certification/)
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:125]: ditto line 5
```suggestion
   [YAML folder](../../src/app/tests/suites/certification/)
```

File: docs/guides/simulated_device_linux.md [hunk 20]
```diff
@@ -0,0 +1,152 @@
...
+2. Test names must follow a strict format dues to CI of test recognition. The
+   format is as follows:
+    - Test_TC\_[`CATEGORY ABBREVIATION`]\_[`SECTION NUMBER`]\_[`SUBSECTION
+      NUMBER`]\_Simulated.yaml
+    - <strong>`IMPORTANT`: The test name must end in Simulated with the
+      capital.</strong>
+3. Available properties can be found in
+   [YAML Test Name](https://github.com/project-chip/connectedhomeip/tree/master/src/app/tests/suites/README.md)
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:133]: ditto line 5
```suggestion
   [YAML Test Name](../../src/app/tests/suites/README.md)
```

File: docs/guides/simulated_device_linux.md [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+4. An Additional property is as follows:
+
+    | Name | Description                                                                 |
+    | ---- | --------------------------------------------------------------------------- |
+    | wait | The command that is expected to be received on the app from the controller. |
+
+5. [Test_TC_DM_1_3_Simulated](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/certification/Test_TC_DM_1_3_Simulated.yaml)
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:140]: ditto line 5
```suggestion
5. [Test_TC_DM_1_3_Simulated](../../src/app/tests/suites/certification/Test_TC_DM_1_3_Simulated.yaml)
```

File: docs/guides/simulated_device_linux.md [hunk 22]
```diff
@@ -0,0 +1,152 @@
...
+   is an example of a written test that runs on the simulated device.
+6. Next, it will need to be added to
+   [examples/placeholder/linux/apps/app1/tests.js](https://github.com/project-chip/connectedhomeip/tree/master/examples/placeholder/linux/apps/app1/tests.js).
```
Comment (electrocucaracha) [inline docs/guides/simulated_device_linux.md:143]: ditto line 5
```suggestion
   [examples/placeholder/linux/apps/app1/tests.js](../../examples/placeholder/linux/apps/app1/tests.js).
```

**Pull request merged**